interceptors.go

```diff
@@ -0,0 +1,23 @@
+package sarama
+
+// ProducerInterceptor allows you to intercept (and possibly mutate) the records
+// received by the producer before they are published to the Kafka cluster.
+// https://cwiki.apache.org/confluence/display/KAFKA/KIP-42%3A+Add+Producer+and+Consumer+Interceptors#KIP42:AddProducerandConsumerInterceptors-Motivation
+type ProducerInterceptor interface {
+
+	// onSend is called when the producer message is intercepted.
+	// Please avoid modifying the message until it's safe to do so, as you'll be
+	// modifying the message reference.
```

Comment:
what's the message reference?

Reply:
What does "until it's safe to do so" mean?

Will that make sense to the user? I'm not sure I know when it's safe 😄

Reply:
you got it here: https://github.com/d1egoaz/sarama/pull/1#discussion_r439616140